From: Nathan Pratta Teodosio <nteodosio@ubuntu.com>
Date: Fri, 27 Sep 2024 13:49:48 +0200
Subject: If /etc/X11/Xresources is a directory

instead of a regular file, load /etc/X11/Xresources/*
into the X resource database

Bug-Ubuntu: https://launchpad.net/bugs/1955850
Forwarded: not-needed
---
 data/Xsession.in | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/data/Xsession.in b/data/Xsession.in
index df9d819..dab24ce 100755
--- a/data/Xsession.in
+++ b/data/Xsession.in
@@ -94,7 +94,7 @@ userresources="$HOME/.Xresources"
 usermodmap="$HOME/.Xmodmap"
 userxkbmap="$HOME/.Xkbmap"
 
-if [ -f /etc/X11/Xresources ]; then
+if [ -e /etc/X11/Xresources ]; then
     sysresources=/etc/X11/Xresources
 else
     sysresources=/usr/etc/X11/Xresources
@@ -120,9 +120,7 @@ if [ -f "$rh6sysresources" ]; then
     xrdb -nocpp -merge "$rh6sysresources"
 fi
 
-if [ -f "$sysresources" ]; then
-    xrdb -nocpp -merge "$sysresources"
-fi
+find "$sysresources" -maxdepth 1 -type f -readable -regextype egrep -regex '.*/[[:alnum:]_-]+$' -exec xrdb -nocpp -merge {} \;
 
 if [ -f "$userresources" ]; then
     xrdb -nocpp -merge "$userresources"
